File: Solutions/0495- Teemo Attacking/Solution.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        total = 0
        endPoison = 0

        for i in range(0,len(timeSeries)):
            endPoison = timeSeries[i] + duration - 1
            if i == len(timeSeries) - 1 or endPoison < timeSeries[i+1]:
                total += duration
            else:
                total += timeSeries[i+1] - timeSeries[i]
        
        return total

File: Solutions/0495- Teemo Attacking/Solution.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int total = 0;
        int endPoison = 0;

        for (int i = 0; i < timeSeries.length; i++) {
            endPoison = timeSeries[i] + duration - 1;
            if (i == timeSeries.length - 1 || endPoison < timeSeries[i+1])
                total += duration;
            else
                total += timeSeries[i+1] - timeSeries[i];
        }

        return total;

    }
}